Adyen/Utilities/Observable/ObservationManager.swift

```diff
@@ -9,8 +9,17 @@ import Foundation
 /// Manages all the observations of an observer.
 internal class ObservationManager {
     
+    private var observations = [Observation]()
+    private let lock = NSLock()
+    
     deinit {
-        observations.forEach {
+        let observationsToRemove = handleWithinLock {
```

Comment:
Did you try to write a unit test to check proper deallocation? Do we even need this deinit?